src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -76,9 +76,12 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
     shareReplay(1),
   );
 
+  private navigationRootElementId$ = new BehaviorSubject<string | undefined>(undefined);
```

Comment:
From what I understand, `navigationRootElementId$` is only used in the context of restricting the navigation (stay "empty" in the regular case). 
If so:
- this (variable + functions) requires good comments in the code what it is for. As it is, it adds a lot of confusion in a service which is already one of the more complex of the app. You also need to explain what `undefined` means for this var.
- not sure `rootElement` is a good name for that. Ok I understand now that you have probably added the `navigation` prefix to show it applies to `navigationNeighbors` only but as everything is in this service in related to navigation, it does not work. I think we need something very explicit here... for instance... `restrictNeighborNavigationToDescendantOfElementId `

Reply:
fixed